webapp/sources/rudder/rudder-web/src/main/elm/onboarding/sources/View.elm

```diff
@@ -0,0 +1,220 @@
+module View exposing (..)
+
+import DataTypes exposing (..)
+import Html exposing (Html, button, div, i, span, text, h1, h3, ul, li, b, label, input, form, a)
+import Html.Attributes exposing (class, type_, name, id, href, for, checked, value)
+import Html.Attributes.Autocomplete as Autocomplete
+import Html.Attributes.Extra exposing (autocomplete)
+import Html.Events exposing (onClick, onInput)
+import List exposing (any, intersperse, map, sortWith)
+import List.Extra exposing (minimumWith)
+import String exposing (lines)
+import Dict exposing (..)
+
+view : Model -> Html Msg
+view model =
+  let
+    completeClass = "section-complete"
+    completeIcon  = "fa fa-check"
+
+    warningClass  = "section-warning"
+    warningIcon   = "fas fa-exclamation"
+
+    defaultClass  = "section-default"
+    defaultIcon   = "fa fa-info"
+
+    visitedClass  = "section-visited"
+
+{--
+    getSelectedMetrics : String
+    getSelectedMetrics =
+      case Dict.get 2 model.sections of
+        Just m -> case m of
+          Metrics _ v -> v
+          _ -> "no"
+        Nothing -> "no"
+--}
+    sidebarSection : Section -> Html Msg
+    sidebarSection s =
+      let
+        titleSection =
+          case s of
+            Welcome _        -> "Welcome"
+            Account _ _      -> "Account"
+            Metrics _ _      -> "Metrics"
+            GettingStarted _ -> "Getting Started"
+
+        sectionIndex = case List.Extra.elemIndex s model.sections of
+          Just i  -> i
+          Nothing -> 0
+        activeClass = if sectionIndex == model.activeSection then "activeSection" else ""
+
+        stateClass  = case s of
+          Welcome _    -> completeClass
+          Account se _ ->
+            case se of
+              Visited  -> visitedClass
+              Complete -> completeClass
+              Warning  -> warningClass
+              _        -> defaultClass
```

Comment:
if _ is only Default please use it instead of _, it will allow the compiler to fail if you add a new State